airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/stream_slicer.py

```diff
@@ -28,3 +28,21 @@ def stream_slices(self) -> Iterable[StreamSlice]:
 
         :return: List of stream slices
         """
+
+    def set_parent_state(self, stream_state: StreamState) -> None:
+        """
+        Set the state of the parent streams.
```

Comment:
the docstring should be more specific about the parameter expectations. stream_state must have `parent_state` which is not obvious from the signature. It might even be worth creating a new class to codify the expectation (kind of like how StreamSlice codifies the presence of a partition and a cursor_slice)

Reply:
Updated docs for all classes